Classes/Driver/Version5/Mapping/NodeTypeMappingBuilder.php

```diff
@@ -0,0 +1,208 @@
+            'POST',
+            '/_scripts/updateFulltextParts',
+            [],
+            json_encode(['script' => [
+                'lang' => 'painless',
+                'code' => '
+                    HashMap fulltext = (ctx._source.containsKey("__fulltext") ? ctx._source.__fulltext : new HashMap());
+                    HashMap fulltextParts = (ctx._source.containsKey("__fulltextParts") ? ctx._source.__fulltextParts : new HashMap());
+                    ctx._source = params.newData;
+                    ctx._source.__fulltext = fulltext;
+                    ctx._source.__fulltextParts = fulltextParts'
+                ]
+            ])
+        );
+
+        $this->client->request(
+            'POST',
+            '/_scripts/regenerateFulltext',
+            [],
+            json_encode(['script' => [
+                'lang' => 'painless',
+                'code' => '
+                        ctx._source.__fulltext = new HashMap();
+                        if (!ctx._source.containsKey("__fulltextParts")) {
+                            ctx._source.__fulltextParts = new HashMap();
+                        }
+                        
+                        if (params.nodeIsRemoved || params.nodeIsHidden || params.fulltext.size() == 0) {
+                            if (ctx._source.__fulltextParts.containsKey(params.identifier)) {
+                                ctx._source.__fulltextParts.remove(params.identifier);
+                            }
+                        } else {
+                            ctx._source.__fulltextParts.put(params.identifier, params.fulltext);
+                        }
+    
+                        ctx._source.__fulltextParts.each {
+                            originNodeIdentifier, partContent -> partContent.each {
+                                bucketKey, content ->
+                                    if (ctx._source.__fulltext.containsKey(bucketKey)) {
+                                        value = ctx._source.__fulltext[bucketKey] + " " + content.trim();
+                                    } else {
+                                        value = content.trim();
+                                    }
+                                    ctx._source.__fulltext[bucketKey] = value;
+                            }
+                        }
+                        '
+                ]
+            ])
+        );
+    }
+
+    /**
+     * Adjust the mapping for string to text or keyword as needed.
+     *
+     * This is used to ease moving from ES 1.x and 2.x to 5.x by migrating the
+     * mapping like this:
+     *
+     * | 2.x                                       | 5.x                              |
+     * |-------------------------------------------|----------------------------------|
+     * | "type": "string", "index": "no"           | "type": "text", "index": false   |
+     * | "type": "string"[, "index": "analyzed"]   | "type": "text", "index": true    |
```

Comment:
Not sure I get that, but in that column, it's always about both, but `analyzed` is the default, so even if not given "it'll be there".